cdk/lib/overwatch-support-construct.ts

```diff
@@ -12,37 +12,34 @@ import {InstallTagFunction} from './install-tag-function';
 import * as path from 'path';
 import * as fs from 'fs';
 import {Stack} from 'aws-cdk-lib';
+import {OverwatchSupportStackProps} from './overwatch-support-stack';
 
 /**
  * Handles the creation of primary services used in Overwatch.
  */
 export class OverwatchSupportConstruct extends Construct {
-  readonly workspace: CfnWorkspace;
+  readonly workspace?: CfnWorkspace;
 
-  constructor(scope: Construct, id: string) {
+  constructor(scope: Construct, id: string, props: OverwatchSupportStackProps) {
     super(scope, id);
 
     const alertsTopic = new AlertsTopic(this, 'AlertsTopic', {
       topicName: 'OverwatchAlerts',
       url: 'https://ingest.centergauge.com/',
     });
 
-    this.workspace = new CfnWorkspace(this, 'Workspace', {
-      alias: 'Overwatch',
-      alertManagerDefinition: fs
-        .readFileSync(
-          path.join(
-            __dirname,
-            '..',
-            '..',
-            'support',
-            'alertmanager.yaml'
-          ),
-          'utf-8'
-        )
-        .replace(/{{{region}}}/g, Stack.of(this).region)
-        .replace(/{{{alertsTopicArn}}}/g, alertsTopic.topic.topicArn),
-    });
+    if (!props.skipAmp) {
+      this.workspace = new CfnWorkspace(this, 'Workspace', {
+        alias: 'Overwatch',
+        alertManagerDefinition: fs
+          .readFileSync(
+            path.join(__dirname, '..', '..', 'support', 'alertmanager.yaml'),
+            'utf-8'
+          )
+          .replace(/{{{region}}}/g, Stack.of(this).region)
+          .replace(/{{{alertsTopicArn}}}/g, alertsTopic.topic.topicArn),
+      });
+    }
```

Comment:
I'm pretty sure these variables are used by the SSM documents also created in this ... we should provide a means to pass an amp URI or have it invalid.. My guess is the SSM doc or running the doc will fail without them

Reply:
If the AMP workspace is not created we also skipping the ssm doc creation here 
```// Conditionally create InstallConstruct only if workspace is available
    if (overwatchSupport.workspace) {
      new InstallConstruct(this, 'Install', {
        workspace: overwatchSupport.workspace,
      });
    } else {
      console.warn(
        'AMP is skipped; skipping the creation of InstallConstruct.'
      );
    }```